x-pack/plugins/security/server/authentication/providers/http.ts

```diff
@@ -79,6 +95,21 @@ export class HTTPAuthenticationProvider extends BaseAuthenticationProvider {
       this.logger.debug(
         `Request to ${request.url.pathname}${request.url.search} has been authenticated via authorization header with "${authorizationHeader.scheme}" scheme.`
       );
+
+      // If Kibana is configured to restrict JWT authentication only to selected routes, ensure that the route is marked
+      // with the `ROUTE_TAG_ACCEPT_JWT` tag to bypass that restriction.
+      if (
+        user.authentication_realm.type === JWT_REALM_TYPE &&
+        this.jwt?.taggedRoutesOnly &&
+        !request.route.options.tags.includes(ROUTE_TAG_ACCEPT_JWT)
+      ) {
+        this.logger.error(
+          `Attempted to authenticate with JWT credentials against ${request.url.pathname}${request.url.search}, but it's not allowed. ` +
+            `Ensure that the route is defined with the "${ROUTE_TAG_ACCEPT_JWT}" tag.`
```

Comment:
**question** What do you think about logging a small portion of the JWT in this message? Thinking about supportability, it might be nice to know if a specific token is being used, and having a fragment of that token in the log message could help with this.

Reply:
Yeah, it's a good idea, I can log some port of the signature (n-last characters, since n-first characters, header, will be the same most of the time).